### PR TITLE
Added in spectrum_from_model

### LINKED ISSUE #337: Create utility method to create a spectrum from a model and spectrum
This is a simple method that is needed given a model (with or without units) and a spectrum, then an output spectrum is created based on the dispersion axis of the input spectrum and the model.

The primary use case could be for building a continuum *spectrum* from a continuum model.

@eteq

### PR DESCRIPTION
**Pull request by brechmos-stsci**
This is a bit of a helper/utility method that creates a Spectrum1D object from a model given an input spectrum. The output Spectrum1D object will have all the same values as the input Spectrum1D except the flux will be based on the input model.

Fixes #337

### CODE REVIEW

File: specutils/manipulation/utils.py [hunk 1]
```diff
@@ -152,3 +152,42 @@ def excise_region(spectrum, region, exciser=linear_exciser):
     #
 
     return exciser(spectrum, region)
+
+
+def spectrum_from_model(model_input, spectrum):
```
Comment (eteq) [inline specutils/manipulation/utils.py]: This needs to be added to `__all__` if it's intended for the public API

File: specutils/manipulation/utils.py [hunk 2]
```diff
@@ -152,3 +152,42 @@ def excise_region(spectrum, region, exciser=linear_exciser):
...
+    other parameters for the output `~specutils.Spectrum1D` object
+    will be the same as the input `~specutils.Spectrum1D` object.
+
+    Parameters
+    ----------
+    model : `~astropy.modeling.Model`
+        The input model or compound model from which flux is calculated.
+
+    spectrum : `~specutils.Spectrum1D`
+        The `~specutils.Spectrum1D` object to which the smoothing will be applied.
```
Comment (eteq) [inline specutils/manipulation/utils.py]: Copy-and-paste error?  Probably should be "The `~specutils.Spectrum1D` object to use as the model template" or something like that.

File: specutils/manipulation/utils.py [hunk 3]
```diff
@@ -152,3 +152,42 @@ def excise_region(spectrum, region, exciser=linear_exciser):
...
+        flux = model_input(spectrum.spectral_axis)
+
+    # If the input model does not have units, then assume it is in
+    # the same units as the input spectrum.
+    else:
+        flux = model_input(spectrum.spectral_axis.value)*spectrum.flux.unit
+
+    return Spectrum1D(flux=flux,
+                      spectral_axis=spectrum.spectral_axis,
+                      uncertainty=spectrum.uncertainty,
```
Comment (eteq) [inline specutils/manipulation/utils.py]: I don't think it makes sense to copy over the uncertainty here - the uncertainty is *not* the same for the model (in one interpretation it's zeros, and in another it's undefined).

(Should also note this in the docstring)

**Comment by eteq** [comment]
A bit more context: this provides an alternative solution to #333 from having the model itself know what to do.  We may want this *anyway*, but its proximate reason is that.

File: specutils/manipulation/utils.py (hunk 1 above)
Comment (brechmos-stsci) [reply specutils/manipulation/utils.py]: 👍

File: specutils/manipulation/utils.py (hunk 2 above)
Comment (brechmos-stsci) [reply specutils/manipulation/utils.py]: 👍

File: specutils/manipulation/utils.py (hunk 3 above)
Comment (brechmos-stsci) [reply specutils/manipulation/utils.py]: 👍

**Comment by eteq** [comment]
Since this isn't critical functionality and we're very close, I'm pushing this to 0.5

**Comment by nmearl** [comment]
@brechmos Can you rebase this?

**Pull request merged**